tests/baselines/reference/assignmentCompatWithDiscriminatedUnion.types

```diff
@@ -359,9 +359,9 @@ namespace GH12052 {
 
     good.type = getAxisType();
 >good.type = getAxisType() : IAxisType
->good.type : IAxisType
+>good.type : "linear" | "categorical"

```

Comment:
This is another interesting case where the alias is no longer preserved. I assume this is because of the *bug* that we'd use a type alias name when we encountered a union with the same constituents? I'd guess this is the *correct* type representation, since the type of `good.type` comes from the union of `ILinearAxis["type"]` and `ICategoricalAxis["type"]`, which wouldn't have an origin.